client/state/selectors/is-eligible-for-signup-destination.js

```diff
@@ -27,23 +13,19 @@ import { retrieveSignupDestination } from 'signup/utils';
  * @return {Boolean} True if current user is able to see the checklist after checkout
  */
 export default function isEligibleForSignupDestination( state, siteId, cart ) {
-	if ( ! isEmpty( getAllCartItems( cart ) ) ) {
-		if (
-			hasDomainMapping( cart ) ||
-			hasDomainRegistration( cart ) ||
-			hasTransferProduct( cart ) ||
-			( ! hasPlan( cart ) && ! hasConciergeSession( cart ) ) ||
-			hasEcommercePlan( cart )
-		) {
-			return false;
-		}
+	if ( hasEcommercePlan( cart ) ) {
+		return false;
 	}
 
 	const destination = retrieveSignupDestination();
 
-	if ( destination && destination.includes( '/checklist/' ) ) {
-		return isNewSite( state, siteId ) && isEligibleForDotcomChecklist( state, siteId );
+	if ( ! destination ) {
+		return false;
+	}
+
+	if ( destination.includes( '/checklist/' ) ) {
```

Comment:
Just curious - do these `includes` calls get transpiled - cause they are not supported in IE? 🤔

Reply:
I believe that's covered by our setup. I use modern features without worrying about browsers support.

- [browserslist](https://github.com/Automattic/wp-calypso/blob/0ecebd1057518f31316043d3dfa844d442e3ede1/package.json#L12-L34)
- [@babel/env](https://github.com/Automattic/wp-calypso/blob/0ecebd1057518f31316043d3dfa844d442e3ede1/babel.config.js#L11-L20)
- [@babel/polyfill](https://github.com/Automattic/wp-calypso/blob/0ecebd1057518f31316043d3dfa844d442e3ede1/client/boot/polyfills.js#L5)

Reply:
Thanks for the pointers! It might not be. I remember coming across this before, just not certain in what context. I guess the answer may be in the browers list (?): 

```
	"last 2 Chrome versions",
	"last 1 ChromeAndroid versions",
	"last 2 Firefox versions",
	"last 2 Safari versions",
	"last 2 iOS versions",
	"last 2 Edge versions",
	"last 2 Opera versions",
	"unreleased Chrome versions",
	"unreleased ChromeAndroid versions",
	"unreleased Firefox versions",
	"unreleased FirefoxAndroid versions",
	"unreleased Edge versions",
	"unreleased Opera versions"
```
So no "IE".

Sorry if spontaneous comment btw (learning 😀).

Reply:
Calypso builds two sets of bundles, one for modern browsers and another for everything else.

The "everything else" gets IE in browserslist via this package config:

https://github.com/WordPress/gutenberg/blob/8a3f3f2659d65a2e017a07bfe8852ef5d61d69bb/packages/browserslist-config/index.js#L3-L4

Reply:
Gotcha - guess I should have followed through to check the defaults! Thanks for clarifying. :-)

(Nice to not worry about any of these)

Reply:
Just to carify - the "evergreen" container should be transpiling for `ie>=11` right?

Reply:
I'm not sure about details there, @sgomes from PerfOps implemented it so he knows. :-)

Reply:
> I'm not sure about details there, @sgomes from PerfOps implemented it so he knows. :-)

That's good - no need. :-)

Reply:
You rang? 😄 

As @simison pointed out, we have two builds: `evergreen`, which has its own config in our `package.json`, and `fallback`, which uses the defaults. The defaults are inherited from `@wordpress/browserslist`, as @simison pointed out too.

At runtime, on the server side, we look at the useragent for incoming requests. If it's a string we recognise as being from a modern browser, and a recent version thereof, we serve it the evergreen files. Otherwise, we serve it the fallback files.

This way, IE11 and any browser that we don't recognise as being modern will always get the fallback bundle, and thus all of the browser support it has built in, namely ES5 transpilation and polyfills for missing features (both manually added polyfills and polyfills added by CoreJS). Only browsers that we know for sure are modern will be sent the leaner, evergreen build, that includes a lot less of this.

In general, this should all be pretty transparent while you're writing code. If the feature you're using is supported by CoreJS v2 (the version we're currently using, and which is mostly language feature-focused), or one of our [manually-added polyfills](https://github.com/Automattic/wp-calypso/blob/master/client/boot/polyfills.js), you should be good to go! 🙂

Reply:
Oh, glad you are here then! Thanks for the generous explanation @sgomes. Very good to know. :-)